src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenManagementService.cs

```diff
@@ -96,13 +107,32 @@ private Func<Task<ClientCredentialsToken>> GetNewTokenTaskFactory(string clientN
 
     private async Task ScheduleTokenBackgroundRefreshAsync(string clientName, TimeSpan delayBeforeNextBackgroundRefresh)
     {
-        await Task.Delay(delayBeforeNextBackgroundRefresh, this._backgroundRefreshCancellationToken).ConfigureAwait(false);
-
         try
         {
-            // We don't care about the token, we just want to refresh it in the background and have it cached
-            _ = await this.GetAccessTokenAsync(clientName, CachingBehavior.ForceRefresh, this._backgroundRefreshCancellationToken).ConfigureAwait(false);
-            Interlocked.Increment(ref this._backgroundRefreshedTokenCount);
+            await Task.Delay(delayBeforeNextBackgroundRefresh, this._backgroundRefreshCancellationToken).ConfigureAwait(false);
+
+            var options = this._optionsMonitor.Get(clientName);
+
+            // This is a background process and must not be associated to the current activity
+            // https://github.com/open-telemetry/opentelemetry-dotnet/issues/984
+            var parentActivity = Activity.Current;
+            Activity.Current = null;
```

Comment:
If we don't create a detached/root trace, all the refresh spans will end up in a single cascading trace.